libs/barista-components/sunburst-chart/src/sunburst-chart.ts

```diff
@@ -135,27 +112,133 @@ export class DtSunburstChart {
     DtSunburstChartNode[]
   > = new EventEmitter();
 
-  /** Template portal for the default overlay */
-  // tslint:disable-next-line: use-default-type-parameter no-any
-  private _portal: TemplatePortal<any> | null;
+  /** @internal Overlay template */
+  @ContentChild(DtSunburstChartOverlay, { static: false, read: TemplateRef })
+  _overlay: TemplateRef<DtSunburstChartTooltipData>;
+
+  /** @internal Viewchild selection of the svg */
+  @ViewChild('svg') _svgEl;
+
+  /** @internal Viewchildren selection for the slices */
+  @ViewChildren(DtSunburstChartSegment) private _segments: QueryList<
+    DtSunburstChartSegment
+  >;
+
+  /** Slices to be painted. Exposed so users can open overlays */
+  get slices(): DtSunburstChartNodeSlice[] {
+    return this._slices;
+  }
+  _slices: DtSunburstChartNodeSlice[] = [];
+
+  /** @internal The chart's width based on the viewport width. */
+  _width = MIN_WIDTH;
+
+  /** @internal The chart's radius. */
+  get _radius(): number {
+    return this._width / 2;
+  }
+
+  // viewbox is originX originY width and height. With -width/2 -height/2 width height the center of the svg is (0,0)
+  // this way we don't have to translate all the elements and it gets centered in the middle
+  /** @internal viewbox representation */
+  get _viewBox(): string {
+    return `${-this._width / 2} ${-this._width / 2} ${this._width} ${
+      this._width
+    }`;
+  }
+
+  /** @internal Label of selected element to be displayed */
+  _selectedLabel: string;
+  /** @internal Value of selected element to be displayed */
+  _selectedValue: number;
+  /** @internal Relative value of selected element to be displayed */
+  _selectedRelativeValue: number;
+  /** @internal Marks if absolute value should be shown or percent instead */
+  _valueAsAbsolute: boolean = true;
+
+  private _filledSeries: DtSunburstChartTooltipData[];
 
   /** Reference to the open overlay. */
-  private _overlayRef: OverlayRef;
+  private _overlayRef: DtOverlayRef<DtSunburstChartTooltipData> | null;
+
+  /** @internal Destroy subject to clear subscriptions on component destroy. */
+  private readonly _destroy$ = new Subject<void>();
 
   constructor(
-    private _overlayService: Overlay,
-    private _viewContainerRef: ViewContainerRef,
+    private _overlayService: DtOverlay,
```

Comment:
I guess this would still be a breaking change, when changing the type from `Overlay` to `DtOverlay`.